### PR TITLE
Refactor sample catalog screenshot production

### PR DESCRIPTION
**Pull request by HansMuller**
The (Java) code used to save screenshots has been flaky on Linux. Refactored the code so that screenshot production can run on (just) MacOS.

Also:
- The screenshot uploader is no longer a separate process.
- ... no longer tries to reuse an HTTP client after a timeout.
- ... logs the files that it has saved.

### CODE REVIEW

**Comment by HansMuller** [comment]
@yjbanov

File: dev/devicelab/lib/tasks/sample_catalog_generator.dart [hunk 1]
```diff
@@ -19,7 +21,8 @@ Future<TaskResult> samplePageCatalogGenerator(String authorizationToken) async {
   await inDirectory(catalogDirectory, () async {
     await flutter('packages', options: <String>['get']);
 
-    if (deviceOperatingSystem == DeviceOperatingSystem.ios)
+    final bool isIOSDevice = deviceOperatingSystem == DeviceOperatingSystem.ios;
```
Comment (yjbanov) [inline dev/devicelab/lib/tasks/sample_catalog_generator.dart]: nit: for consistency with other code, let's treat "ios" as a plain word and camel-case it like this: `isIosDevice`.

File: dev/devicelab/pubspec.yaml [hunk 2]
```diff
@@ -9,6 +9,7 @@ environment:
 
 dependencies:
   args: ^0.13.4
+  image:
```
Comment (yjbanov) [inline dev/devicelab/pubspec.yaml]: In most other places we're more specific about the version. How about `^1.1.27`?

**Review by yjbanov — APPROVED** [review]
LGTM

**Pull request merged**